File: frontend/src/data/weather.ts
import type { WeatherCondition } from '@/types';
import weatherJson from '@data/weather.json';

/**
 * 날씨 시나리오 — 공유 데이터셋(data/weather.json). 검증(기획서 §8)과 데모 토글에 사용.
 */
export type WeatherScenarioId = 'normal' | 'heatwave' | 'coldwave' | 'rain' | 'dust';

export const WEATHER_SCENARIOS = weatherJson as unknown as Record<
  WeatherScenarioId,
  WeatherCondition
>;

export const DEFAULT_WEATHER: WeatherScenarioId = 'normal';

File: backend/app/data/weather.py
"""날씨 시나리오 — 공유 데이터셋(data/ai/weather.json)에서 로드."""
from __future__ import annotations

from ..models import WeatherCondition
from ._loader import load

WEATHER_SCENARIOS: dict[str, WeatherCondition] = {
    k: WeatherCondition.model_validate(v) for k, v in load("weather.json").items()
}

DEFAULT_WEATHER = "normal"


def get_weather(scenario: str | None) -> WeatherCondition:
    return WEATHER_SCENARIOS.get(scenario or DEFAULT_WEATHER, WEATHER_SCENARIOS[DEFAULT_WEATHER])
